### ISSUE TITLE
textOutput() and gridOutput() seems to always label a canvas as white regardless of actual background color

### ISSUE THREAD

**Issue by limzykenneth** [processing/p5.js#6610]
### Most appropriate sub-area of p5.js?

- [X] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

1.9.0

### Snippet:

```js
function setup() {
  createCanvas(400, 400);
}

function draw() {
  background("blue");
  textOutput();
}

```

Gives inner HTML of canvas element of 
```html
<div id="defaultCanvas0accessibleOutput" role="region" aria-label="Canvas Outputs">
  <div id="defaultCanvas0textOutput">
    Text Output
    <div id="defaultCanvas0textOutputSummary" aria-label="text output summary">
      <p id="defaultCanvas0textOutput_summary">
        Your output is a, 400 by 400 pixels, white canvas containing the following 0 shapes:
      </p>
      <ul id="defaultCanvas0textOutput_list"></ul>
    </div>
    <table id="defaultCanvas0textOutput_shapeDetails" summary="text output shape details"></table>
  </div>
</div>
```

**Comment by pbelokon** [comment]
@limzykenneth  Can I have a try ?

**Comment by ujjwaleee26** [comment]
@limzykenneth  I understood the issue ,actually according to the available code , colorText="white" is defined inside the loop , which makes to prone to get selected even before checking the colourExceptions array completely. Declaring it outside the loop  and If a match is found at any point in the loop, the loop is exited using break, and 'white' is not assigned. This ensures that the final value of colortext accurately reflects whether a match was found in the colorExceptions array.  Can you please assign this issue to me .

**Comment by ujjwaleee26** [comment]
Hey @limzykenneth 
There are cases in your issue, If you use `textOutput(LABEL)` in `function setup()` you will get correct color written, and if you will in `function draw()` you will get white canvas given if `background(color) `is used in `function setup` .I am attaching 2 screen shots for the 2 cases-. I figured a solution for the following , declaration of `colortext=white` outside the loop, so that first comparison is done the `colorException` array completely and white is not prematurely assigned . Can you please assign this issue to me, I really want to contribute and learn the working of the code .
<img width="1680" alt="Screenshot 2023-12-09 at 4 14 06 PM" src="https://github.com/processing/p5.js/assets/138270264/e02435ab-65be-4bc4-9939-4097adb9d17c">
<img width="1680" alt="Screenshot 2023-12-09 at 4 11 21 PM" src="https://github.com/processing/p5.js/assets/138270264/1d39e621-4708-43db-98cf-e64589c3e435">

**Comment by limzykenneth** [comment]
Hi @ujjwaleee26, that is not where the issue is. For example with the following sketch
```js
function setup() {
  createCanvas(400, 400);
  textOutput(LABEL);
}

function draw() {
  background(frameCount % 255);
}
```
The text label for the background should have gone from "black" to "grey" to "white". Preassiging the value or not does not really matter if the background color text label does not change when the background color change.

**Comment by ujjwaleee26** [comment]
Hey @limzykenneth ,
I understood that you want a dynamic color rendering feature which can track changeColorChange. I would love to work on it . Can I give it a try?

**Comment by ujjwaleee26** [comment]
Hey @limzykenneth ,
Is there any canvasId which can be used to generalise the change?

**Comment by limzykenneth** [comment]
@ujjwaleee26 I don't think that is relevant. 

I have found the cause of the problem and will create a fix shortly.

**Comment by ujjwaleee26** [comment]
Het @limzykenneth ,
I have found the way of dynamically rendering color name, but there will be some logical changes , like when no background is there in reference to `defaultCanvas0` it should show transparent(black) `rgb(0,0,0)` instead of white and it works, Can you let me have a try?

**Comment by limzykenneth** [comment]
@ujjwaleee26 That is not where the problem with this issue is. The problem is that changes in background color is not kept in track in the system. #6619 will fix this problem.

**Comment by ujjwaleee26** [comment]
Hey @limzykenneth 
I understood that issue, that's why only I created a way of dynamically rendering it. I am sure your solution will be correct cause you have a good understanding of codebase than I do. Can you explain your solution in a bit more detail like what you thought, and for once consider my solution too.

**Comment by limzykenneth** [comment]
The problem in this issue is not about dynamic color, that has always been in p5.js Web Accessibility and it is not broken nor require implementation. The problem can be identified when testing the code path that are run or not run when the background is set to a specific color. The color white while it is the default color is also not a problem and is not in the wrong position in the loop, if the background is a different color it would have overwritten that value already.

Your solution using `getComputedStyle` gets the canvas element's CSS background color which is not what the `background` function is setting in this case so there is no reason to implement a function for that in the first place. Do have a read on the PR's description and the code changes to see what the actual problem in this issue is, which is more of a change tracking problem and not a color problem.

My personal recommendation is to really spend some time understanding the system and how it works before even thinking about any potential solution. Create test cases, add breakpoints/use console logging message to find out different code path's behavior, understand why any particular part of the code is written that way and what it is trying to do, all these can really help work out where the problem lies and only then should a potential solution be found.
